libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -269,6 +269,24 @@ bool SyntaxChecker::visit(VariableDeclaration const& _declaration)
 	return true;
 }
 
+bool SyntaxChecker::visit(VariableDeclarationStatement const& _statement)
+{
+	// Check if declaration contains at least one named component, and report otherwise.
+	size_t const namedVariablesCount = count_if(
+		_statement.declarations().begin(),
+		_statement.declarations().end(),
+		[](ASTPointer<VariableDeclaration> const& decl) { return decl.get() != nullptr; }
+	);
+
+	if (namedVariablesCount == 0)
+		m_errorReporter.syntaxError(
+			_statement.location(),
+			"Use of the \"var\" keyword is disallowed. Variable declaration statement also does not declare any variables."
```

Comment:
Maybe
``The use of the "var" keyword is disallowed. The statement does not declare any variables, so the empty "var"-declaration on the left hand side can be removed.`` or something similar? Also the source location should probably be the declaration part of the statement, not the whole statement.

Reply:
With the correct source location, maybe ``so the empty declaration part of the statement can be removed.`` also works. Maybe you find something that's still better :).

Reply:
Or ``The use of the "var" keyword is disallowed. The declaration part of the statement can be removed, since it is empty.``

Reply:
@chriseth any objections towards the very last suggestion from @ekpyron ?

Reply:
Sounds good! (not that anyone will read that message... ;) )